Exiled.Events/Patches/Events/Scp096/TryingNotToCry.cs

```diff
@@ -0,0 +1,48 @@
+// -----------------------------------------------------------------------
+// <copyright file="TryingNotToCry.cs" company="Exiled Team">
+// Copyright (c) Exiled Team. All rights reserved.
+// Licensed under the CC BY-SA 3.0 license.
+// </copyright>
+// -----------------------------------------------------------------------
+
+namespace Exiled.Events.Patches.Events.Scp096
+{
+#pragma warning disable SA1313
+    using Exiled.Events.EventArgs;
+
+    using HarmonyLib;
+
+    using Interactables.Interobjects.DoorUtils;
+
+    using PlayableScps;
+
+    using UnityEngine;
+
+    /// <summary>
+    /// Patches the <see cref="Scp096.TryNotToCry"/> method.
+    /// Adds the <see cref="Handlers.Scp096.TryingNotToCry"/> event.
+    /// </summary>
+    [HarmonyPatch(typeof(Scp096), nameof(Scp096.Charging))]
+    internal static class TryingNotToCry
+    {
+        private static bool Prefix(Scp096 __instance)
+        {
+            if (!Physics.Raycast(__instance.Hub.PlayerCameraReference.position, __instance.Hub.PlayerCameraReference.forward, out RaycastHit hitInfo, 1f, LayerMask.GetMask("Door")))
```

Comment:
You could internally cache the layermask to use instead of calling GetMask every time.

Reply:
I think that's the base-game code.